libminifi/src/FlowController.cpp

```diff
@@ -431,8 +431,9 @@ int16_t FlowController::clearConnection(const std::string &connection) {
   return -1;
 }
 
-std::shared_ptr<state::response::ResponseNode> FlowController::getAgentManifest() const {
+std::shared_ptr<state::response::ResponseNode> FlowController::getAgentManifest() {
   auto agentInfo = std::make_shared<state::response::AgentInformation>("agentInfo");
+  agentInfo->setUpdatePolicyController(std::static_pointer_cast<controllers::UpdatePolicyControllerService>(getControllerService(c2::C2Agent::UPDATE_NAME)));
```

Comment:
Nonconst getters could  lead to some bad practices in my opinion.

Since we only use this UpdatePolicyControllerService to call const functions (namely canUpdate()) I think these update_policy_controller_ members could be made into a `shared_ptr<const controllers::UpdatePolicyControllerService>`
And if we were to create a bunch of boilerplate to enable getting a shared_ptr<const ControllerServiceNode> from a const FlowController, then we could keep getAgentManifest as a const function.

Reply:
Good point, I was a bit hesitant changing it because it was very low core level stuff and I thought it would be convoluted, but after trying it, it seems these controller provider getters could be made const quite easily. Updated it in 5e37b8b76b97c6e04ee145c80293b5557348ba66